components/eamxx/src/diagnostics/tests/aodvis_test.cpp

```diff
@@ -0,0 +1,106 @@
+                                        const int nlevs) {
+  const int num_global_cols = ncols * comm.size();
+
+  using vos_t = std::vector<std::string>;
+  ekat::ParameterList gm_params;
+  gm_params.set("grids_names", vos_t{"Point Grid"});
+  auto &pl = gm_params.sublist("Point Grid");
+  pl.set<std::string>("type", "point_grid");
+  pl.set("aliases", vos_t{"Physics"});
+  pl.set<int>("number_of_global_columns", num_global_cols);
+  pl.set<int>("number_of_vertical_levels", nlevs);
+
+  auto gm = create_mesh_free_grids_manager(comm, gm_params);
+  gm->build_grids();
+
+  return gm;
+}
+
+TEST_CASE("aodvis") {
+  using namespace ShortFieldTagsNames;
+  using namespace ekat::units;
+
+  // A world comm
+  ekat::Comm comm(MPI_COMM_WORLD);
+
+  // A time stamp
+  util::TimeStamp t0({2022, 1, 1}, {0, 0, 0});
+
+  const auto nondim = Units::nondimensional();
+
+  // Create a grids manager - single column for these tests
+  constexpr int nlevs = 33;
+  const int ngcols    = 2 * comm.size();
+
+  // TODO: Don't hardcode this!
+  constexpr int nbnds = 14;
+
+  auto gm   = create_gm(comm, ngcols, nlevs);
+  auto grid = gm->get_grid("Physics");
+
+  // Input (randomized) tau
+  FieldLayout scalar3d_swband_layout{{COL, SWBND, LEV}, {ngcols, nbnds, nlevs}};
+  FieldIdentifier tau_fid("aero_tau_sw", scalar3d_swband_layout, nondim,
+                          grid->name());
+  Field tau(tau_fid);
+  tau.allocate_view();
+  tau.get_header().get_tracking().update_time_stamp(t0);
+
+  // Construct random number generator stuff
+  using RPDF = std::uniform_real_distribution<Real>;
+  RPDF pdf(0, 0.05);
+  auto engine = scream::setup_random_test();
+
+  // Construct the Diagnostics
+  std::map<std::string, std::shared_ptr<AtmosphereDiagnostic>> diags;
+  auto &diag_factory = AtmosphereDiagnosticFactory::instance();
+  register_diagnostics();
+
+  constexpr int ntests = 5;
+  for(int itest = 0; itest < ntests; ++itest) {
+    // Randomize tau
+    randomize(tau, engine, pdf);
+
+    // Create and set up the diagnostic
+    ekat::ParameterList params;
+    auto diag = diag_factory.create("AerosolOpticalDepth550nm", comm, params);
+    diag->set_grids(gm);
+    diag->set_required_field(tau);
+    diag->initialize(t0, RunType::Initial);
+
+    // Run diag
+    diag->compute_diagnostic();
+
+    // Check result
+    tau.sync_to_host();
+    diag->get_diagnostic().sync_to_host();
+
+    const auto tau_h  = tau.get_view<const Real ***, Host>();
+    const auto aod_hf = diag->get_diagnostic();
+
+    Field aod_tf = diag->get_diagnostic().clone();
+    aod_tf.deep_copy<double, Host>(0.0);
+    auto aod_t = aod_tf.get_view<Real *, Host>();
+
+    for(int icol = 0; icol < grid->get_num_local_dofs(); ++icol) {
+      for(int ilev = 0; ilev < nlevs; ++ilev) {
+        // TODO: Don't hardcode the 10!
+        aod_t(icol) += tau_h(icol, 10, ilev);
+      }
+    }
+    aod_hf.sync_to_dev();
+    aod_tf.sync_to_dev();
+    REQUIRE(views_are_equal(aod_hf, aod_tf));
```

Comment:
The only places we use `view_reduction()` all tests are guarded by `SCREAM_BFB_TESTING`, so there isn't this issue in release mode. A possible workaround is the following, which would mean we aren't testing the diagnostic in release, which I'd be ok with. Maybe @bartgol has other thoughts?
```suggestion
    // Workaround for non-bfb behavior of view_reduction() in release builds
    if (SCREAM_BFB_TESTING) { 
      REQUIRE(views_are_equal(aod_hf, aod_tf));
    }
```